test/e2e/routes.test.ts

```diff
@@ -0,0 +1,128 @@
+import { describe, test, expect } from "./baseFixture"
+import { clean, tmpdir } from "../utils/helpers"
+import * as path from "path"
+import { promises as fs } from "fs"
+
+const routes = ["/", "/vscode", "/vscode/"]
+
+describe("VS Code Routes", ["--disable-workspace-trust"], {}, async () => {
+  const testName = "integrated-terminal"
+  test.beforeAll(async () => {
+    await clean(testName)
+  })
+
+  test("should load all route variations", async ({ codeServerPage }) => {
+    for (const route of routes) {
+      await codeServerPage.navigate(route)
+
+      // Check there were no redirections
+      const url = new URL(codeServerPage.page.url())
+      expect(url.pathname).toBe(route)
+
+      // Check that page loaded from correct route
+      const html = await codeServerPage.page.innerHTML("html")
+      switch (route) {
+        case "/":
+        case "/vscode/":
+          expect(html).toMatch(/src="\.\/[a-z]+-[0-9a-z]+\/static\//)
+          break
+        case "/vscode":
+          expect(html).toMatch(/src="\.\/vscode\/[a-z]+-[0-9a-z]+\/static\//)
+          break
+      }
```

Comment:
Just a thought (not something we need to do in this PR or even ever since this works just fine): now that we are in a proper browser instead of scraping the HTML we could possibly intercept requests to make sure assets are loading from the right spot.

Reply:
Good call! I'll add a comment to the code for now.